pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -409,6 +414,62 @@ func buildPeerSecurityRule(rulePriority *int32, protoName armnetwork.SecurityRul
 	return securityRule
 }
 
+// findSecurityRule finds the security rule in the given slice and return the index with boolean indicating found or not.
+func findSecurityRule(rule *armnetwork.SecurityRule, ruleList []*armnetwork.SecurityRule) (int, bool) {
```

Comment:
This function not only finds the rule, it also checks if the rule is changed.
Please change the function name accordingly.

Reply:
You mean check the rule list is changed? I don't think so, changing the rule in list to nil is just an optimization, the goal is still to find if the rule exist in the list.